manifests/init.pp

```diff
@@ -438,6 +480,123 @@
       enable  => $enable_pcsd_service,
       require => Package['pcs'],
     }
+
+    # Determine if this node should perform authorizations
+    case $manage_pcsd_auth_node {
+      'first': { $auth_node = $quorum_members[0] }
+      'last': { $auth_node = $quorum_members[length($quorum_members)-1] }
+      default: {}
+    }
+
+    # If the local data matches auth_node (hostname or primary IP) we can
+    # perform auth processing for subsequent components
+    if $trusted['certname'] == $auth_node or
+        $facts['networking']['ip'] == $auth_node {
+          $is_auth_node = true
+    } else {
+      $is_auth_node = false
+    }
+
+    $exec_path = '/sbin:/bin/:usr/sbin:/usr/bin'
+
+    if $manage_pcsd_auth and $is_auth_node {
+      if ! $sensitive_hacluster_password or ! $sensitive_hacluster_hash {
+        fail('The hacluster password and hash must be provided to authorize nodes via pcsd.')
+      }
+
+      # TODO - verify if this breaks out of the sensitivity
+      $hacluster_password = $sensitive_hacluster_password.unwrap
+      $auth_credential_string = "-u hacluster -p ${hacluster_password}"
```

Comment:
That sums it up. \The simple solution would be to remove the "sensitive" flag. Would that be acceptable for the moment?

My only thought on preserving sensitivity would be to create a provider for pcs auth so the parameter can be passed to a resource. Any other suggestions?

Reply:
I'm not sure if I'm just misunderstanding `Sensitive` right now, but it only removes the content from the catalog and the debugging output; you should not *need* to unwrap sensitive as it should just be pasted correctly into the `exec`, unless I'm entirely mistaken? (Haven't tested it.)

Reply:
Unfortunately if you try to substitute a sensitive value into an non-sensitive string it actually inserts the "Sensitive string redacted" message instead of the content. I ended up putting that comment in during testing after trying exactly what you're suggesting here.